### ISSUE TITLE
"ignore: use_setters_to_change_properties" doesn't work with a preceding dartdoc comment

### ISSUE THREAD

**Issue by jamesderlin** [dart-lang/linter#2430]
With `use_setters_to_change_properties` enabled in my `analysis_options.yaml`, running `dart analyze` against the following code:

```dart
class Foo {
  int member = 0;

  /// Documentation goes here.
  // ignore: use_setters_to_change_properties
  void setMember(int value) {
    member = value;
  }
}
```

gives me:

> info • Use a setter for operations that conceptually change a property at foo.dart:4:3 • (use_setters_to_change_properties)

I find this surprising since the `ignore:` line does work if I either move it above the dartdoc comment or if I remove the dartdoc comment completely.  It seems strange to me that this somehow seems sensitive to the placement of the dartdoc comment (and if anything, I'm even more surprised that it works with the intervening dartdoc comment).

I'm using Dart SDK version: 2.12.353624752-edge+google3-v2 (google3) (google3) on "linux_x64".

**Comment by srawlins** [comment]
Agree. The diagnostic is being reported on the entire method, but should only be reported on the name. You get 5 lines of squiggly lines as is.
